node/src/types/block.rs

```diff
@@ -783,14 +778,54 @@ impl From<bytesrepr::Error> for BlockValidationError {
     }
 }
 
+/// A storage representation of finality signatures with the associated block hash.
+#[derive(Debug, Serialize, Deserialize, Clone, DataSize)]
+pub struct BlockSignatures {
+    /// The block hash for a given block.
+    pub block_hash: BlockHash,
+    /// The era id for the given set of finality signatures
+    pub era_id: EraId,
+    /// The signatures associated with the block hash.
+    pub proofs: BTreeMap<PublicKey, Signature>,
```

Comment:
These fields should ideally not be `pub`.  Instead there should be a `pub(crate)` getter for each so that we can be sure the only way a `BlockSignatures` struct can be modified by a caller is via its one mutable method `append_proof()`.

Reply:
Changed to `pub(crate)`

Reply:
I meant that we should have these member variables private, and have `pub(crate)` methods like `pub(crate) fn block_hash(&self) -> &BlockHash`